File: FMAplication/Domain/Products/POSMProduct.cs
using FMAplication.Attributes;
using FMAplication.Core;
using FMAplication.Domain.Brand;
using System.ComponentModel.DataAnnotations;

namespace FMAplication.Domain.Products
{
    public class POSMProduct : AuditableEntity<int>
    {
        protected POSMProduct()
        {
            IsJTIProduct = true;
            IsDigitalSignatureEnable = true;

        }

        [Required]
        [StringLength(256)]
        public string Name { get; set; }

        [Required]
        public int Type { get; set; }

        public bool IsJTIProduct { get; set; }

        public bool IsDigitalSignatureEnable { get; set; }
        public bool IsPlanogram { get; set; }
        public string PlanogramImageUrl { get; set; }
        public string ImageUrl { get; set; }

        [UniqueKey(groupId: "1", order: 0)]
        [Required]
        [StringLength(128, MinimumLength = 3)]
        public string Code { get; set; }

        public int? BrandId { get; set; }
        public int? SubBrandId { get; set; }
        public int CampaignId { get; set; }
        public FMAplication.Domain.Brand.Brand Brand { get; set; }
        public SubBrand SubBrand { get; set; }
        public FMAplication.Domain.Campaign.Campaign Campaign { get; set; }

    }
}

File: FMAplication/ClientApp/src/app/Shared/Entity/Products/posmproduct.ts
﻿import { Status } from "../../Enums/status";
import { YesNo } from "../../Enums/yesno";

export class PosmProduct {
    public id: number;
    public name: string;
    public type: number;
    public isJTIProduct: boolean;
    public isDigitalSignatureEnable: boolean;
    public isPlanogram: boolean;
    public planogramImageUrl: string;
    public imageFile:File; 
    public imageUrl : string;
    public status: number;
    public code: string;
    public planogramImageFile: File;
    public campaignId: number;
    public brandId: number;
    public subBrandId: number;

    //for client side
    isJTIProductLabel: string;
    isSelected: boolean;
    isDigitalSignatureEnableLabel: string;

    constructor() {
        this.id = 0;
        this.name = '';
        //this.isJTIProduct = YesNo.Yes;
        //this.isDigitalSignatureEnable = YesNo.Yes;
        this.status = Status.Active;
        this.isSelected = false;
    }
}